src/Build/Evaluation/Evaluator.cs

```diff
@@ -1988,6 +1993,11 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
                                 _lastModifiedProject = importedProjectElement;
                             }
 
+                            if (importedProjectElement.StreamTime != null && importedProjectElement.StreamTime > _lastModifiedProject.LastWriteTimeWhenRead)
+                            {
+                                _streamImports.Append(";").Append(importedProjectElement.FullPath);
```

Comment:
I think it would be less allocatey to keep a list of elements and do a `Join(';')` at the end.